examples/rest-api.graphql

```diff
@@ -1,12 +1,13 @@
 schema
   @server(port: 8000, graphiql: true, hostname: "0.0.0.0")
-  @upstream(baseURL: "http://jsonplaceholder.typicode.com", httpCache: true)
-  @link(type: Operation, src: "operations/routes.graphql") {
+  @upstream(onRequest: "onRequest", baseURL: "http://jsonplaceholder.typicode.com", httpCache: true)
+  @link(type: Operation, src: "operations/routes.graphql")
+  @link(type: Script, src: "scripts/echo.js") {
   query: Query
 }
 
 type Query {
-  posts: [Post] @http(path: "/posts")
+  posts: [Post] @http(path: "/posts", onRequest: "postsRequestHandler")
```

Comment:
Instead of making changes here, can you add some tests, like we have in the tests folder for JS.

Reply:
Sure.
Please point the tests folder location.

Reply:
Check this test out for example: https://github.com/tailcallhq/tailcall/blob/main/tests/execution/test-js-request-reponse.md

Reply:
@tusharmath  
added a test